packages/legend-studio/src/components/editor/side-bar/Explorer.tsx

```diff
@@ -300,7 +304,9 @@ const PackageTreeNodeContainer = observer(
 
     return (
       <ContextMenu
-        content={<ExplorerContextMenu node={node} />}
+        content={
+          <ExplorerContextMenu node={node} nodeIsImmutable={readOnlyContext} />
```

Comment:
why don't we add this as property of the node?

Reply:
more work I guess, but probably cleaner than this. `isImmutable` or `isReadOnly` on the node itself might be useful for other things as well.